rpc/client.go

```diff
@@ -26,212 +26,606 @@ type Transport interface {
 	Close() error
 }
 
-type Client interface {
-	Logger() log.Logger
-	ServerConfig() ServerConfig
-	User() discord.User
-	V() int
-	Transport() Transport
-
-	Subscribe(event Event, args CmdArgs, handler Handler) error
-	Unsubscribe(event Event, args CmdArgs) error
-	Send(message Message, handler Handler) error
-	Close()
+type responseMessage struct {
+	data MessageData
+	err  error
 }
 
 func NewClient(clientID snowflake.ID, opts ...ConfigOpt) (Client, error) {
 	config := DefaultConfig()
 	config.Apply(opts)
 
-	client := &clientImpl{
-		logger:          config.Logger,
+	client := Client{
+		Logger:          config.Logger,
 		eventHandlers:   map[Event]Handler{},
-		commandHandlers: map[string]internalHandler{},
+		commandChannels: map[string]chan responseMessage{},
 		readyChan:       make(chan struct{}, 1),
+		clientID:        clientID,
 	}
 
 	if config.Transport == nil {
 		var err error
 		config.Transport, err = config.TransportCreate(clientID, config.Origin)
 		if err != nil {
-			return nil, err
+			return Client{}, err
 		}
 	}
-	client.transport = config.Transport
+	client.Transport = config.Transport
+
+	return client, nil
+}
 
-	go client.listen(client.transport)
+type Client struct {
+	Logger    log.Logger
+	Transport Transport
+
+	eventHandlers   map[Event]Handler
+	commandChannels map[string]chan responseMessage
+	clientID        snowflake.ID
+
+	readyChan    chan struct{}
+	User         discord.User
+	ServerConfig ServerConfig
+	V            int
+}
+
+func (c *Client) Open() error {
+	go c.listen(c.Transport)
 
 	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
 	defer cancel()
 	select {
 	case <-ctx.Done():
-		return nil, ctx.Err()
-	case <-client.readyChan:
+		return ctx.Err()
+	case <-c.readyChan:
 	}
+	return nil
+}
 
-	return client, nil
+func (c *Client) sendBytes(r io.Reader) error {
```

Comment:
this has only one usage. why is this here?

Reply:
in old rpc branch this was the same (`(c *clientImpl) send(r io.Reader) error`) with only one usage. This has been inlined in [f531263](https://github.com/dennisrijsdijk/disgo/commit/f531263bd0a55277578b3e5055aef04f5bcab81d)